Empty Stats: Reader Discover nudge spotlight view

momo-ozawa opened the pull request:
Part of #17126 

## Description
This PR adds the spotlight view (pulsing blue circle) to the Reader Discover nudge flow.

Follow Topics | Reader Discover
-- | --
<img src="https://user-images.githubusercontent.com/6711616/138744738-8d7efe5c-5a66-4d75-b9ff-9ceefe8b178c.png" width=200> | <img src="https://user-images.githubusercontent.com/6711616/138744731-0e4d5cf3-a466-46f6-bfcd-5aba81409b1c.png" width=200>

## How to test
1. Go to **My Site** > **Stats** for site with less than 30 views
2. Complete or dismiss the first two nudges (**Publicize** and **Blogging reminders**) in order to have the **Reader discover** nudge shown
3. Tap on **Discover blogs to follow**
4. **Follow topics** screen should be presented
5. ✅ Notice the spotlight view in the center of the screen
6. Select one or more topics and tap **Done**
7. **Reader** tab should be presented, with **Discover** tab selected
8. ✅ Notice the spotlight view on top of the comments button on the first post

## Regression Notes
1. Potential unintended areas of impact
Follow topics screen, Reader Discover screen

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the above steps

3. What automated tests I added (or what prevented me from doing so)
N/A, just visual changes

## PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Commit:
Add: spotlight view for Followed Topics screen

Commit:
Add: spotlight view for Reader Discover

Commit:
Add: display notice on Discover screen

momo-ozawa commented:
@osullivanchris 👋 

Lmk if the following is okay:
- The position of the spotlight view on Reader Discover
- The toast message copy on Reader Discover 

I've approved the workflow for an installable build. I think you should be able to download from [this comment ](https://github.com/wordpress-mobile/WordPress-iOS/pull/17354#issuecomment-949799326 )once the build completes 😄

osullivanchris commented:
@momo-ozawa 

- position looks good 👍 
- toast message also looks correct 👍 

A couple of things I noticed (but might not be finished yet):
- when I tap on a tag, I think the spotlight should dismiss. Currently its still there as I'm tapping on other tags.
- I went and tapped the comment icon on a post in discover. I didn't actually comment, but I went back. The spotlight was still there on the comment icon. I think it should be gone once I tap on it once.
- I went through the flow again for a second site. On this occasion, it didn't prompt me to comment. It only gave me the first tag step.
- I like that the loading copy says 'Following new topics' as it implies the user might already follow some topics. I'm not 100% sold on the location of the loading state. But its also not high priority.

osullivanchris reviewed (commented):
forgot to tap the green 'review changes' button

Commit:
Fix: hide spotlight when tag is tapped

Commit:
Fix: hide spotlight when comment button is tapped

momo-ozawa commented:
@osullivanchris @leandroalonso @nikola-milicevic Ready for a re-review 😄

https://user-images.githubusercontent.com/6711616/138896868-1b8bbd68-0eb5-4d1e-bd42-77f79c880ca1.mp4

> when I tap on a tag, I think the spotlight should dismiss. Currently its still there as I'm tapping on other tags.

Fixed! 26017fc

> I went and tapped the comment icon on a post in discover. I didn't actually comment, but I went back. The spotlight was still there on the comment icon. I think it should be gone once I tap on it once.

Fixed! dfddc4f

> I went through the flow again for a second site. On this occasion, it didn't prompt me to comment. It only gave me the first tag step.

I couldn't reproduce this 😢 

> I like that the loading copy says 'Following new topics' as it implies the user might already follow some topics. I'm not 100% sold on the location of the loading state. But its also not high priority.

We can update this in a different PR if needed!

_Note: Installable build available [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/17354#issuecomment-949799326)_

Commit:
Refactor: use exisiting ReaderPostCellDelegate to handle spotlight view

osullivanchris commented:
Those two fixes look perfect per the video @momo-ozawa  🙌

leandroalonso commented on WordPress/Classes/ViewRelated/Reader/ReaderPostCardCell.swift:
```diff
@@ -244,6 +263,16 @@ private extension ReaderPostCardCell {
         menuButton.setImage(highlightIcon, for: .highlighted)
     }
 
+    func setupSpotlightView() {
```
Any specific reason for this to be public?

momo-ozawa replied:
Fixed! 494ebe2

leandroalonso commented on WordPress/Classes/ViewRelated/Reader/Tab Navigation/ReaderTabViewController.swift:
```diff
@@ -104,6 +104,7 @@ class ReaderTabViewController: UIViewController {
 
     func presentDiscoverTab() {
         let discoverIndex = 1
```
This wasn't part of your changes but I think we can add that as a static constant in `ReaderStreamViewController`, that way if this index changes in the future this won't be affected.

momo-ozawa replied:
Fixed! e8632f0

leandroalonso commented on WordPress/Classes/ViewRelated/Reader/Tab Navigation/WPTabBarController+ReaderTabNavigation.swift:
```diff
@@ -41,7 +41,9 @@ extension WPTabBarController {
         if content.topicType == .discover, let topic = content.topic {
-            return ReaderCardsStreamViewController.controller(topic: topic)
+            let controller = ReaderCardsStreamViewController.controller(topic: topic)
+            controller.isReaderDiscoverNudgeFlow = readerTabViewModel.isReaderDiscoverNudgeFlow
```
This part is mainly to display the spotlight icon on the comment button, right?

I wonder if we should be more practical and rename it to `shouldShowCommentSpotlight` or something like that. I'm pretty sure someone else just reading `isReaderDiscoverNudgeFlow` won't be able to guess for what that is.

momo-ozawa replied:
Fixed! 48f858b

leandroalonso reviewed (approved):
@momo-ozawa feature-wise: all good! Left a few comments to improve the code. :)

Commit:
Fix: make setupSpotlightView() private

Commit:
Refactor: define discover index as a static constant

Commit:
Refactor: rename isReaderDiscoverNudgeFlow to shouldShowCommentSpotlight

momo-ozawa commented:
@leandroalonso @nikola-milicevic 
I've addressed the comments above -- feel free to merge this PR! 🙇‍♀️

nikola-milicevic commented:
Tested, works as expected, merging it. Thanks @momo-ozawa @osullivanchris @leandroalonso!
